client/lib/wp/sync-handler/index.js

```diff
@@ -178,22 +180,49 @@ export class SyncHandler {
 			key = new Hashes.SHA1().hex( key );
 		}
 
+		key = 'sync-record-' + key;
+
 		debug( 'key: %o', key );
 		return key;
 	}
 
-	retrieveRecord( key, fn = () => {} ) {
+	retrieveRecord( key ) {
 		debug( 'getting data from %o key\n', key );
-		return localforage.getItem( key, fn );
+		return localforage.getItem( key );
 	}
 
-	storeRecord( key, data, fn = () => {} ) {
+	/**
+	 * Add/Override a record.
+	 *
+	 * @param {String} key - record key identifier
+	 * @param {Object} data - data to store
+	 * @return {Promise} native ES6 promise
+	 */
+	storeRecord( key, data ) {
 		debug( 'storing data in %o key\n', key );
-		return localforage.setItem( key, data, fn );
+		return cache
+			.addItem( key )
+			.then( localforage.setItem( key, data ) );
 	}
 
-	removeRecord( key, fn = () => {} ) {
+	removeRecord( key ) {
 		debug( 'removing %o key\n', key );
-		return localforage.removeItem( key, fn );
+		return cache
+			.removeItem( key )
+			.then( localforage.removeItem( key ) );
```

Comment:
I can understand the logic in `storeRecord` where we don't add the new record until we've added the key by which we can prune it. We want to avoid having a record without a key in the list. But if that's our purpose, then the order by which we _remove_ records should be reversed... we should remove the record first and then only remove the key once we confirm the record has been removed.